graylog2-web-interface/src/components/common/EntityDataTable/types.ts

```diff
@@ -25,12 +25,14 @@ export type Column = {
   type?: boolean,
 };
 
+// A column render should have either a `width` and optionally a `minWidth` or only a `staticWidth`.
 export type ColumnRenderer<Entity extends { id: string }> = {
-  maxWidth?: string,
   renderCell?: (entity: Entity, column: Column) => React.ReactNode,
   renderHeader?: (column: Column) => React.ReactNode,
   textAlign?: string,
-  width?: string,
+  minWidth?: number, // px
+  width?: number, // fraction of unassigned table width, similar to CSS unit fr.
+  staticWidth?: number, // px
```

Comment:
I tried to define the type in a way that you can only I tried to define the type in a way that you can only deine a `staticWidth` or a `width` and optionally a `minWidth`, but it is problematic because we are merging the custom and default column definition. Especially the custom column definitions provided by e.g. the `StreamsOverview` are often not complete.